modules/gloo/features/feature.go

```diff
@@ -38,7 +55,57 @@ func Register(ctx context.Context, rContext *types.Context) error {
 			ingress.Register,
 			setting.Register,
 		},
+		OnStart: func() error {
+			// handle upgrade
+			return upgradeOldVirtualService(rContext.Namespace, rContext.RestConfig)
+		},
 	}
 
 	return feature.Register()
 }
+
+// upgradeOldVirtualService handles compatibility from upgrading gloo from v0.20.x to v1.0.0 for virtualservices.
+func upgradeOldVirtualService(systemNamespace string, config *rest.Config) error {
```

Comment:
This says upgrade, but really it just deletes them so they can be recreated ?

Reply:
Corrrect, delete non-compatible types